x-pack/plugins/observability_solution/apm/server/routes/assets/services/get_assets.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { Logger } from '@kbn/core/server';
+import { errors } from '@elastic/elasticsearch';
+import { WrappedElasticsearchClientError } from '@kbn/observability-plugin/server';
+import { termQuery, kqlQuery, rangeQuery } from '@kbn/observability-plugin/server';
+import { ASSET_TYPE, FIRST_SEEN, LAST_SEEN } from '../../../../common/es_fields/assets';
+import { AssetsESClient } from '../../../lib/helpers/create_es_client/create_assets_es_client/create_assets_es_clients';
+import { withApmSpan } from '../../../utils/with_apm_span';
+import { ServiceAssetDocument } from './types';
+
+export const MAX_NUMBER_OF_SERVICES = 1_000;
+
+type AssetType = 'service';
+
+export function assetsRangeQuery(start: number, end: number): QueryDslQueryContainer[] {
+  return [
+    {
+      range: {
+        [FIRST_SEEN]: {
+          gte: start,
+        },
+      },
+    },
+    {
+      range: {
+        [LAST_SEEN]: {
+          lte: end,
+        },
+      },
+    },
+  ];
+}
+
+export async function getAssets({
+  assetsESClient,
+  start,
+  end,
+  kuery,
+  assetType,
+  logger,
+}: {
+  assetsESClient: AssetsESClient;
+  start: number;
+  end: number;
+  kuery: string;
+  assetType: AssetType;
+  logger: Logger;
+}) {
+  return withApmSpan('get_assets', async () => {
+    try {
+      const response = await assetsESClient.search(`get_${assetType}_from_assets`, {
+        body: {
+          size: MAX_NUMBER_OF_SERVICES,
+          track_total_hits: false,
+          query: {
+            bool: {
+              filter: [
+                ...termQuery(ASSET_TYPE, assetType),
+                ...kqlQuery(kuery),
+                ...assetsRangeQuery(start, end),
+              ],
+            },
+          },
+        },
+      });
+
+      const services = response.hits.hits.map((hit) => {
```

Comment:
These are two different things IMO.

We should have a `getAssets` function which will perform the query you wrote and another one `getServiceAssets` which will call the `getAssets` passing `assetType=service` and where you would convert the result into a service list like on line 73. 

Do you think it makes sense?

Another option would be to kind of roll back to your first version with a function called `getServiceAssets` but this function doesn't need to receive an `assetType` argument. You would hard coded it in the filter query.

But maybe we should not stress ourselves with this at this point. We don't know yet if this will be reusable when we move it to the right plugin.

Reply:
> We should have a getAssets function which will perform the query you wrote and another one getServiceAssets which will call the getAssets passing assetType=service and where you would convert the result into a service list like on line 73.

I understand your point. It does make sense that it's more abstract, with each function having a single responsibility.

It's a fairly easy change and with the tests in place it's even easier. Let me update it

Reply:
https://github.com/elastic/kibana/pull/181842/commits/5185031883a51f0b6e132c017239aa669853b6d7